### ISSUE TITLE
Fallback to file completions if custom/external completions are empty

### ISSUE THREAD

**Issue by ysthakur** [nushell/nushell#14595]
### Related problem

When making a custom completer, there's no obvious way to use Nushell's built in file completions for certain arguments. This is a problem brought up in https://github.com/nushell/nushell/issues/14582.

And when using external completers, some completers don't always provide completions. For example, Carapace doesn't provide file completions when I type `go run <TAB>`. I won't bother adding steps for reproducing because this particular problem might just be me, but I imagine it would also make other external completers' lives easier if they could shell out to Nushell for file completions.

### Describe the solution you'd like

If either a custom completer or the external completer don't provide any completions, we should provide file completions instead.

The only drawback I can think of is that this may confuse users in some cases if they weren't expecting file completions, but I think that's fine.

### Describe alternatives you've considered

Currently, if an external completer doesn't return a list, we fall back to file completions. So, users can set things up so that if their external completer returns an empty list, they return `null` or something. However, this feels pretty hacky, and it doesn't work for custom completions.

**Comment by NotTheDr01ds** [comment]
I thought this was working at one point, but I may have been mistaken.

If we change this, then I think:

* Returning `null` from the completer should allow fallback to file-completions
* Returning an empty list from the completer should suppress completions

**Comment by ysthakur** [comment]
I thought about this a bit more, and sure, it also makes sense to just modify custom completers to fall back to file completions on invalid values rather than on empty lists and to leave external completers alone. I guess if a completer returns an empty list, providing file completions may not be helpful either. And this'd be an even simpler change, we'd just need to modify [this bit](https://github.com/nushell/nushell/blob/main/crates/nu-cli/src/completions/custom_completions.rs#L116) here so a `None` is returned on invalid values.

However, @NotTheDr01ds, I'm not sure if you meant we'd fall back to file completions specifically when custom completers return `null`, but I would prefer it if we did this for any value that isn't a record or list.

@oderwat Thoughts?

**Comment by oderwat** [comment]
@ysthakur, I prefer a fallback mechanism that only applies to "defined" values. This means falling back to null and generating an error (with no completion) for invalid values, as it simplifies debugging. However, I'm uncertain about how this impacts the user experience.

Additionally, I was considering whether it might be beneficial to expose the file completer as a function. This function could even provide "default values" for non-existent files (like "default.config" or other specified files) and allow for filtering options (such as directory-only or specific patterns). Implementing this could be a viable alternative to returning null.

Thinking of how Nushell works, this could be structured as a proper Nushell value that can be filtered and processed. This approach would also enable us to define the styling for completions and include example implementations and data to illustrate how it works.

**Comment by ysthakur** [comment]
@oderwat Yeah, showing errors for invalid values would be best, I'm just now sure how easy it would be. I'll look into that.

Having builtin commands like `complete file` (and maybe `complete command`, `complete flag`, etc.) would indeed be nice, and it might not actually be that complicated to implement. But I'm not sure how often you would need to filter/customize your file completions. I feel like most people would just want all the file completions without customization, and the same thing would be done by the fall-back-to-file-completions-on-null approach.

For displaying files in general (not just for completions), we could perhaps add a new command to read `$env.LS_COLORS` and use that to get the style for a particular file's metadata. That would give you more control over the completions you show.

**Comment by oderwat** [comment]
I just wrote [gunner] (https://github.com/oderwat/gunner) and that would like to have only "*.gleam" files. I have some others that would be "better" when filtering for other file endings or even full names. But if we can create those lists and haven them colored I think one can just create some helper functions and use those.
